eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -261,17 +263,26 @@ private void handleWriteRowsEvent(Event event, Optional<BinlogFileOffset> starti
 
       BinlogEntry entry = extractor.extract(schemaAndTable, eventData, binlogFilename, offset);
 
-
       if (!shouldSkipEntry(startingBinlogFileOffset, entry.getBinlogFileOffset())) {
         binlogEntryHandlers
-              .stream()
-              .filter(bh -> bh.isFor(schemaAndTable))
-              .forEach(binlogEntryHandler -> binlogEntryHandler.publish(entry));
+                .stream()
+                .filter(bh -> bh.isFor(schemaAndTable))
+                .forEach(binlogEntryHandler -> publish(entry, binlogEntryHandler, event));
       }
     }
 
     onEventReceived();
-    saveOffset(event);
+  }
+
+  private void publish(BinlogEntry entry, BinlogEntryHandler binlogEntryHandler, Event event) {
+    CompletableFuture<?> future = binlogEntryHandler.publish(entry);
+
+    offsetProcessor.saveOffset(future, extractBinlogFileOffset(event));
+  }
+
+  private BinlogFileOffset extractBinlogFileOffset(Event event) {
+    offset = extractOffset(event);
```

Comment:
Why is `offset` not a local variable!

Reply:
It seems before there was some reason, but now I also do not see it.
So I will move it to local variable.